tests/unit_tests/test_mesh_from_domain.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import openmc
+import pytest
+
+
+def test_mesh_from_cell():
```

Comment:
How would you feel about naming these `test_reg_mesh_*`? I could see us adding support this kind of method on other mesh types in the future.

Reply:
that is a better name, thanks for the suggestion.